authorization/role/repository/identity_role.go

```diff
@@ -654,3 +697,136 @@ func identityRoleFilterByIdentityID(identityID uuid.UUID) func(db *gorm.DB) *gor
 		return db.Where("identity_id = ?", identityID)
 	}
 }
+
+// FlagStaleForIdentityRoleChange executes two update queries; the first sets the stale flag to true for all privilege cache records where
+// the identity ID is equal to, or a descendent of (via memberships) the specified identity ID, and the resourceID is
+// equal to, or a descendent of (via the resource hierarchy) the specified resource ID.
+// The second query updates the token table, setting the STALE flag of the token STATUS field to true, for all
+// token records that are mapped to the corresponding privilege cache records in the first query, via the
+// many-to-many TOKEN_PRIVILEGE table
+func (m *GormIdentityRoleRepository) FlagPrivilegeCacheStaleForIdentityRoleChange(ctx context.Context, identityID uuid.UUID, resourceID string) error {
```

Comment:
same question as above about the location of this method here, instead of in the privilege cache repository.

Reply:
See previous response.